sqlglot/dialects/tsql.py

```diff
@@ -324,6 +324,34 @@ def _parse(args: t.List[exp.Expression]) -> exp.Expression:
     return _parse
 
 
+# https://learn.microsoft.com/en-us/sql/t-sql/functions/parsename-transact-sql?view=sql-server-ver16
+def _build_parsename(args: t.List) -> t.Optional[exp.SplitPart]:
+    if len(args) != 2:
+        return None
+    arg_this: exp.Literal = seq_get(args, 0) or exp.Literal.string("")
+    arg_partnum: exp.Literal = seq_get(args, 1) or exp.Literal.number(1)
+    text = arg_this.this
+    part_num = int(arg_partnum.this)
+    length = 1 if isinstance(arg_this, exp.Null) else len(text.split(".")) + 1  # Reverse index
+    idx = 0 if isinstance(arg_this, exp.Null) else int(part_num)
```

Comment:
@VaggelisD I still leave NULL there because it's supported. I know that it's weird if using NULL, but the syntax still valid though.

Reply:
Is it worth addressing, though? Otherwise it's adding unnecessary code complexity for an edge case. Consider this:

- Spark
```SQL
spark-sql (default)> select split_part(NULL, '.', 1);
split_part(NULL, ., 1)
NULL

spark-sql (default)> select split_part(NULL, '.', 4);
split_part(NULL, ., 4)
NULL
```

- TSQL:
```SQL
1> select parsename(NULL, 1)
2> go
NULL


1> select parsename(NULL, 4)
2> go
NULL
```

The same goes for DuckDB and I imagine other engines as well; So even if you leave the index as it is you'd still get `NULL`.